htmx-spring-boot/src/main/java/io/github/wimdeblauwe/htmx/spring/boot/mvc/HtmxHandlerInterceptor.java

```diff
@@ -30,22 +29,22 @@ public HtmxHandlerInterceptor(ObjectMapper objectMapper, HtmxResponseHandlerMeth
     @Override
     public void postHandle(HttpServletRequest request, HttpServletResponse response, Object handler, ModelAndView modelAndView) throws Exception {
         if (modelAndView != null) {
-            modelAndView.getModel().values().forEach(
-                    value -> {
-                        if (value instanceof HtmxResponse) {
-                            buildAndRender((HtmxResponse) value, modelAndView, request, response);
-                        } else if (value instanceof HtmxResponse.Builder) {
-                            buildAndRender(((HtmxResponse.Builder) value).build(), modelAndView, request, response);
-                        }
-                    });
+            for (Object value : modelAndView.getModel().values()) {
+                if (value instanceof HtmxResponse res) {
+                    buildAndRender(res, modelAndView, request, response);
+                } else if (value instanceof HtmxResponse.Builder builder) {
+                    buildAndRender(builder.build(), modelAndView, request, response);
+                }
+            }
         }
     }
 
     private void buildAndRender(HtmxResponse htmxResponse, ModelAndView mav, HttpServletRequest request, HttpServletResponse response) {
         View v = htmxResponseHandlerMethodReturnValueHandler.toView(htmxResponse);
         try {
             v.render(mav.getModel(), request, response);
-            htmxResponseHandlerMethodReturnValueHandler.addHxHeaders(htmxResponse, response);
+            // ModelAndViewContainer is not available here, so flash attributes won't work
```

Comment:
Not sure when exactly the flash attributes will not work? Maybe we should document this in the README?

Reply:
That's what I pointed out in my comment above.